code/modules/chemistry/Reagents-Misc.dm

```diff
@@ -698,6 +698,76 @@ datum
 					W.icon = I
 				return
 
+		graphene
+			name = "graphene"
+			id = "graphene"
+			description = "A miniscule honeycomb lattice."
+			reagent_state = SOLID
+			fluid_r = 20
+			fluid_g = 20
+			fluid_b = 20
+			value = 5
+
+		graphene_compound
+			name = "Graphene Hardening Compound"
+			id = "graphene_compound"
+			description = "A thick industrial compound used to reinforce things."
+			reagent_state = LIQUID
+			fluid_r = 10
+			fluid_g = 10
+			fluid_b = 10
+			transparency = 180
+			viscosity = 0.8
+			value = 9
+
+			reaction_obj(var/obj/O, var/volume)
+				if (volume < 1)
+					return
+
+				var/colorize
+				if (istype(O,/obj/machinery/atmospherics/pipe/simple))
+					var/obj/machinery/atmospherics/pipe/simple/P = O
+
+					if(P.can_rupture)
+						var/max_reinforcement = 1e9
+						if(P.fatigue_pressure >= max_reinforcement)
+							return
+
+						P.fatigue_pressure = clamp(P.fatigue_pressure + ((P.fatigue_pressure) * (2**volume)), initial(P.fatigue_pressure), max_reinforcement)
```

Comment:
math here feels a little funky - probably don't need the `P.fatigue_pressure + `, since `x * 2^y >= x` for all non-negative `y`, and you're clamping it just in case anyways 🤷‍♀️

Reply:
I think I was trying to not punish folks that ended up with having smaller unit values but it already checks for units greater than 1 I agree that is more intuitive.